### PR TITLE
Update semconv for prometheus components

### LINKED ISSUE #32814: Prometheus receiver uses Deprecated attributes from old semconv version
### Component(s)

receiver/prometheus

### Is your feature request related to a problem? Please describe.

The Prometheus receiver [uses semconv/v1.6.1 since 2 years ago](https://github.com/open-telemetry/opentelemetry-collector-contrib/blame/main/receiver/prometheusreceiver/internal/prom_to_otlp.go#L12)
to add resource attributes to convert from the Prometheus to the OpenMetrics convention [when scraping](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/prometheusreceiver/internal/prom_to_otlp.go#L42-L48)
but these are confusing in meaning and deprecated according to [latest documentation](https://github.com/open-telemetry/semantic-conventions/blob/5d7e48746a1f01059b86bc59f62c4daa09f86903/docs/attributes-registry/network.md#deprecated-network-attributes).

In addition, there are many dependencies (alerts, dashboards, recording rules) in the world which depend on the job and instance labels in metrics but the receiver is dropping these without any option to keep them.

### Describe the solution you'd like

The attributes should be updated to match the spec [described here](https://opentelemetry.io/docs/specs/otel/compatibility/prometheus_and_openmetrics/#resource-attributes).

Dropping job and instance attributes in the receiver should be configurable so Prometheus labels can be kept in addition to the semconv labels added. Or this should not be done by receivers at all as it can be done be attribute processor.

### Describe alternatives you've considered

Currently using the attributes processor to re-add the job and instance attributes, and to remove the confusing net.* attributes.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dashpole**
**Description:**

Fixes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32814

It places the removal of replaced resource attributes (`net.host.name`, `net.host.port`, and `http.scheme`) behind a feature gate, `receiver.prometheusreceiver.RemoveLegacyResourceAttributes`, which is initially disabled by default.

This means the current default behavior is to write BOTH new and old semconv.  When the feature gate moves to beta, the default will be only new semconv, but with the ability for users to revert to having both.  When the feature gate moves to GA, it will mean users only get the new semconv.

### CODE REVIEW

**Review by TylerHelmuth — CHANGES_REQUESTED** [review]
Upgrading semconv versions terrifies me because of all the breaking changes that come with it, especially jumping from 1.6 to 1.22.

@dashpole can you verify that the only breaking changes with this update are the 3 mentioned in your changelog? 

I'm also don't think we should do this breaking change in 1 go.  For other breaking changes around names, like metric scrapping controlled by mdatagen, we are able to emit both old and new values for some time.  The SDKs were able to do this as well for the stable http semconv break change.  Is that technique something we can do here?

**Review by codeboten — COMMENTED** [review]
This change was merged if you want to pull the changes in https://github.com/open-telemetry/opentelemetry-collector/pull/10075

**Comment by dashpole** [comment]
These are the other semantic conventions referenced:

* service.name
* service.namespace
* service.instance.id
* k8s.pod.name
* k8s.pod.uid
* k8s.container.name
* k8s.node.name
* k8s.replicaset.name
* k8s.daemonset.name
* k8s.statefulset.name
* k8s.job.name
* k8s.cronjob.name
* k8s.namespace.name

None of those have changed since v1.6.1 of the semantic conventions.

**Comment by dashpole** [comment]
@TylerHelmuth this is no longer a breaking change.  It unconditionally adds the new semconv, and adds a feature gate, `receiver.prometheusreceiver.RemoveLegacyResourceAttributes`, which removes the old semconv.  Since it is disabled by default to start, it won't break anyone.

**Comment by TylerHelmuth** [comment]
> None of those have changed since v1.6.1 of the semantic conventions.

🚀 

>  this is no longer a breaking change. It unconditionally adds the new semconv, and adds a feature gate, receiver.prometheusreceiver.RemoveLegacyResourceAttributes, which removes the old semconv. Since it is disabled by default to start, it won't break anyone.

❤️ 

I will give it another in-depth review next week

File: receiver/prometheusreceiver/internal/prom_to_otlp_test.go [hunk 1]
```diff
@@ -57,76 +59,151 @@ func makeK8sResource(jobInstance *jobInstanceDefinition, def *k8sResourceDefinit
...
+	attrs := resource.Attributes()
+	// Using hardcoded values to assert on outward expectations so that
+	// when variables change, these tests will fail and we'll have reports.
+	attrs.PutStr("service.name", def.job)
+	if hasHost {
+		attrs.PutStr("server.address", def.host)
+	}
+	attrs.PutStr("service.instance.id", def.instance)
+	attrs.PutStr("server.port", def.port)
+	attrs.PutStr("url.scheme", def.scheme)
```
Comment (robincw) [inline receiver/prometheusreceiver/internal/prom_to_otlp_test.go:72]: What happens if the scraped metrics already have these attributes, has this been discussed when developing the spec? Shouldn't the resource attributes for a scrape endpoint be in the `telemetry.*` prefix instead? Sorry, this is the wrong place to discuss spec, but maybe the right people are here for s start

Comment (dashpole) [reply receiver/prometheusreceiver/internal/prom_to_otlp_test.go:72]: Today, it isn't possible for prometheus metrics to have attributes which have dots, so collisions are not currently possible.

File: receiver/prometheusreceiver/internal/prom_to_otlp.go [hunk 2]
```diff
@@ -8,8 +8,20 @@ import (
...
+	conventions "go.opentelemetry.io/collector/semconv/v1.25.0"
+	oldconventions "go.opentelemetry.io/collector/semconv/v1.6.1"
+)
+
+const removeOldSemconvFeatureGateID = "receiver.prometheusreceiver.RemoveLegacyResourceAttributes"
+
+var removeOldSemconvFeatureGate = featuregate.GlobalRegistry().MustRegister(
+	removeOldSemconvFeatureGateID,
+	featuregate.StageAlpha,
+	featuregate.WithRegisterFromVersion("v0.100.0"),
```
Comment (TylerHelmuth) [inline receiver/prometheusreceiver/internal/prom_to_otlp.go:22]:
```suggestion
	featuregate.WithRegisterFromVersion("v0.101.0"),
```

**Review by TylerHelmuth — APPROVED** [review]

File: receiver/prometheusreceiver/internal/prom_to_otlp_test.go (hunk 1 above)
Comment (dashpole) [reply receiver/prometheusreceiver/internal/prom_to_otlp_test.go:72]: The `target_info` prometheus metric is converted to the OpenTelmetry resource, so there will be the possibility of collision in the future.  In that case, I think we should use `honor_labels` in the prom config to determine which takes precedence.

Comment (dashpole) [reply receiver/prometheusreceiver/internal/prom_to_otlp_test.go:72]: I do wonder if these "general attributes" should actually be added as resource attributes at all...

Comment (dashpole) [reply receiver/prometheusreceiver/internal/prom_to_otlp_test.go:72]: Resolving, as I don't think this blocks this PR.

File: .chloggen/prometheus-semconv-update.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: 'enhancement'
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: prometheusreciever
```
Comment (bertysentry) [inline .chloggen/prometheus-semconv-update.yaml:7]: typo:

```suggestion
component: prometheusreceiver
```

**Review by djaglowski — APPROVED** [review]

**Pull request merged**